src/HexagonOptimize.cpp

```diff
@@ -678,6 +678,19 @@ class OptimizePatterns : public IRMutator {
                 expr = mutate(op->a + neg_b);
                 return;
             } else {
+
+                // No widening subtracts for the expressions of the form:
+                //      cast<int16_t>(wild_i8x) - cast<int16_t>(wild_i8x)
+                //      cast<int16_t>(wild_u8x) - cast<int16_t>(wild_u8x)
+                // Convert such expressions to vmpa.
+                if (op->type.bits() == 16 && op->type.is_int() &&
+                    ((lossless_cast(UInt(8, op->type.lanes()), op->a).defined() && lossless_cast(UInt(8, op->type.lanes()), op->b).defined()) ||
```

Comment:
Like this?
`                if (op->type.bits() == 16 && op->type.is_int() && 
                    ((lossless_cast(UInt(8, op->type.lanes()), op->a).defined() &&
                    lossless_cast(UInt(8, op->type.lanes()), op->b).defined()) ||
                    (lossless_cast(Int(8, op->type.lanes()), op->a).defined() && 
                    lossless_cast(Int(8, op->type.lanes()), op->b).defined()))) {`